### PR TITLE
Add Snowflake Data Connector

### PR DESCRIPTION
**Pull request by sgrebnov**
Add Snowflake Data Connector

### CODE REVIEW

File: snowflake/README.md [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+
+```yaml
+version: v1beta1
+kind: Spicepod
+name: snowflake-app
+datasets:
+- from: snowflake:snowflake_sample_data.tpch_sf1.lineitem
+  name: lineitem
+  params: 
+    account: ztyvqyb-vgb35424
```
Comment (digadeesh) [inline snowflake/README.md]: better to put a placeholder here

File: snowflake/README.md [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
```
Comment (digadeesh) [inline snowflake/README.md]: put placeholders here.

File: snowflake/README.md (hunk 1 above)
Comment (sgrebnov) [reply snowflake/README.md]: Updated, thanks!

File: snowflake/README.md (hunk 2 above)
Comment (sgrebnov) [reply snowflake/README.md]: Updated, thanks!

File: snowflake/README.md [hunk 3]
```diff
@@ -0,0 +1,209 @@
+# Quickstart for Snowflake Data Connector
+
+>[Snowflake](https://www.snowflake.com/) is a leading cloud-based data warehousing service that enables users to store, compute, and analyze vast amounts of data in real-time.
+
+The guide demonstrates how to configure [Spice with Snowflake Data Connector](https://docs.spiceai.org/data-connectors/snowflake) to access a Snowflake database.
+
+The guide require Snowflake account. Start [free trial](https://signup.snowflake.com/) if needed.
```
Comment (y-f-u) [inline snowflake/README.md]:
```suggestion
The guide requires a Snowflake account. Start [free trial](https://signup.snowflake.com/) if needed.
```

**Review by y-f-u — APPROVED** [review]

**Pull request merged**